### DISCUSSION TITLE
Distributed vector/array of real values

### DISCUSSION THREAD

**Discussion by DaveLanday** [spacejam/rust-crdt#147] [Q&A] [answered]
I am trying to parallelize an enumeration based algorithm. It involves traversing a tree where each node needs information from the parent. The information in question is a vector of real values. It doesn't grow in size (ie: the vector has a fixed size at every node in the tree based on the initialization at the root node). In a shared memory environment, this doesn't seem an issue, but if I want to split work on another machine, I need a copy of the vector on that machine and if the vector is large then it will be hard to send it over a network. Is there a CRDT that is appropriate for this use case? Could I use a merkle dag? Does this discussion belong else-where?

**Comment by davidrusu** [comment, marked as answer]
I don't think CRDT's will help you here. Some suggestions:
-  if each node only needs a subset of the vector, then see if you can partition the vector before sending it to the child nodes?
- design the topology of the network so that all the vector distribution happens early on, i.e. the first few levels of the tree end up copying the vector to other machines, but the rest of the tree is processed locally on each node

**Reply by DaveLanday** [reply]
I see. Thank you @davidrusu for the insight.
